### DISCUSSION TITLE
Always provide desired sizes in media library formats

### DISCUSSION THREAD

**Discussion by Istanful** [strapi/strapi#8694] [Legacy Topics]
Hi!

I recently released a new version of my site where I utilise the medium format provided by the media library. To my surprise it was sometimes missing.

It turns out that some formats are not generated if Strapi deems it unnecessary. Although I understand the sentiment I think that it makes for a so-so API. We now have to add a fallback in case the desired format is not present.

How about always providing `small`, `medium` and `large`? In the case that the image is too small to be in a certain size I propose you use the original url instead. That way the API stays consistent and extra image processing isn't made.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
